src/block.c

```diff
@@ -2047,9 +2047,10 @@ coap_block_test_q_block(coap_session_t *session, coap_pdu_t *actual) {
                                           (0 << 4) | (0 << 3) | 0),
                      buf);
   set_block_mode_probe_q(session->block_mode);
-  coap_delete_pdu(session->saved_pdu);
-  session->saved_pdu = actual;
-  return coap_send_internal(session, pdu);
+  if ((mid = coap_send_internal(session, pdu)) == COAP_INVALID_MID)
```

Comment:
Updated with suggestion

Reply:
Better. But now it reads:

```c
if (x)
  return x;
do_y();
return x;
```

Which looks a bit weird. But feel free to keep it this way since it is neither wrong nor overly complex.